### ISSUE TITLE
device-sdk-c's rest value type is different from go-mod-core-contracts.

### ISSUE THREAD

**Issue by kaisawind** [edgexfoundry/device-sdk-c#303]
The uppercase and lowercase letters of the value type string are different from `go-mod-core-contracts`.
Leading to a wrong identification of the value type of the mixed service of go and c.

https://github.com/edgexfoundry/device-sdk-c/blob/706af55e05cb2cdc0073fc9ec2061f045ee0028b/src/c/edgex-rest.c#L169-L175

https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/models/reading.go
```go
// Constants related to Reading ValueTypes
const (
	ValueTypeBool         = "Bool"
	ValueTypeString       = "String"
	ValueTypeUint8        = "Uint8"
	ValueTypeUint16       = "Uint16"
	ValueTypeUint32       = "Uint32"
	ValueTypeUint64       = "Uint64"
	ValueTypeInt8         = "Int8"
	ValueTypeInt16        = "Int16"
	ValueTypeInt32        = "Int32"
	ValueTypeInt64        = "Int64"
	ValueTypeFloat32      = "Float32"
	ValueTypeFloat64      = "Float64"
	ValueTypeBinary       = "Binary"
	ValueTypeBoolArray    = "BoolArray"
	ValueTypeStringArray  = "StringArray"
	ValueTypeUint8Array   = "Uint8Array"
	ValueTypeUint16Array  = "Uint16Array"
	ValueTypeUint32Array  = "Uint32Array"
	ValueTypeUint64Array  = "Uint64Array"
	ValueTypeInt8Array    = "Int8Array"
	ValueTypeInt16Array   = "Int16Array"
	ValueTypeInt32Array   = "Int32Array"
	ValueTypeInt64Array   = "Int64Array"
	ValueTypeFloat32Array = "Float32Array"
	ValueTypeFloat64Array = "Float64Array"
)
```

**Comment by iain-anderson** [comment]
For v2: change these constants to mixed-case as per core-contracts, enforce correct case usage (drop use of strcasecmp), adjust any sample/example profiles and documentation accordingly.

Historical note: It was agreed in the WG on Sept 14th that the specification would change to match the Go implementation as shown above.

**Comment by kaisawind** [comment]
> For v2: change these constants to mixed-case as per core-contracts, enforce correct case usage (drop use of strcasecmp), adjust any sample/example profiles and documentation accordingly.
> 
> Historical note: It was agreed in the WG on Sept 14th that the specification would change to match the Go implementation as shown above.

thank you very much.

**Comment by iain-anderson** [comment]
closed #321
